ext/dom/node.c

```diff
@@ -329,6 +329,21 @@ int dom_node_node_value_write(dom_object *obj, zval *newval)
 		return FAILURE;
 	}
 
+	switch (nodep->type) {
+		case XML_ELEMENT_NODE:
+		case XML_ATTRIBUTE_NODE:
+		case XML_TEXT_NODE:
+		case XML_COMMENT_NODE:
+		case XML_CDATA_SECTION_NODE:
+		case XML_PI_NODE:
```

Comment:
Is it possible to move this logic inside dom_node_is_read_only?

Reply:
> Is it possible to move this logic inside dom_node_is_read_only?

Unfortunately not, per the description of `nodeValue` in the [DOM Level 3 Node IDL](https://www.w3.org/TR/DOM-Level-3-Core/core.html#ID-1950641247):

> The value of this node, depending on its type; see the table above. When it is defined to be `null`, setting it has no effect, including if the node is read-only.
> `NO_MODIFICATION_ALLOWED_ERR`: Raised when the node is readonly and if it is not defined to be `null`.

For node types where `nodeValue` is `null` by definition, setting `nodeValue` has no effect, not even throwing exceptions (there's nothing to fail); at least that's my reading of it.  Do you agree with my interpretation?

If so:  `dom_node_is_read_only()` doesn't know which property you're trying to set or method you're trying to call, so it can't handle this distinction.